### ISSUE TITLE
[doc] Gloo Mesh resource samples should use service ns instead of gloo-mesh

### ISSUE THREAD

**Issue by linsun** [solo-io/gloo-mesh#1442]
**Describe the bug**
Throughout many places in our docs, we show users to deploy Gloo Mesh custom resources in Gloo-mesh ns.  I don't think this is good practice and contradict what we asks users to do in istio.  Typically istio resources are in source service namespace or destination service namespace.

Another reason this is bad practice is that users can't use sidecar resource to segregate configs pushed to envoy by namespace easily.

Examples: https://docs.solo.io/gloo-mesh/main/guides/multicluster_communication/ - should be deploying to the bookinfo ns.

**Comment by rvennam** [comment]
hm...if I apply TrafficPolicies in the gloo-mesh namespace in the mgmt clsuter, the generated Istio config..like VirtualServices end up in the source service namespace in the workload cluster.

**Comment by djannot** [comment]
Yes, exactly. And we provide our own RBAC which is not based on where the objects are created.
So, I would say that the namespace used doesn't really matter.
